src/printer.js

```diff
@@ -775,13 +780,14 @@ function genericPrintNoParens(path, options, print) {
             concat([
               line,
               "? ",
-              indent(options.tabWidth, path.call(print, "consequent")),
+              indent(options.tabWidth, printedConsequent),
               line,
               ": ",
-              indent(options.tabWidth, path.call(print, "alternate"))
+              indent(options.tabWidth, printedAlternate)
             ])
           )
-        ])
+        ]),
+        { shouldBreak: willBreak(printedConsequent) || willBreak(printedAlternate) }
```

Comment:
I'm not sure how this fixes it because none of the examples I see in the tests have hard lines. They are all expressions that we try to collapse.

`willBreak` just looks to see if there are hard lines, and only really need to be used when using  `conditionalGroup`, because that doesn't automatically get broken if anything inside it has `breakParent`. Otherwise, hard lines emit a `breakParent` which should propagate breaks to all parent groups. This group be getting that behavior automatically.

Which part of the code inside arrays and conditionals are the things that have hard lines, and why isn't it the automatic break propagation working?